horovod/torch/adapter_v2.h

```diff
@@ -17,6 +17,7 @@
 #define HOROVOD_TORCH_ADAPTER_V2_H
 
 #include <torch/extension.h>
+#include <torch/torch.h>
```

Comment:
I still see a deprecation warning with `torch.h`:
```
   #warning "Including torch/torch.h for C++ extensions is deprecated. Please include torch/extension.h"
```

Is it going to continue working for the foreseeable future?  Is the plan to move those missing declarations to `torch/extension.h`, at which point we can remove `torch/torch.h` include with a version guard?

Reply:
I'm going to make a PR very soon to make `torch/extension.h` the canonical header for all C++ extensions and includes everything from `torch/torch.h` too. The warnings will then go away too.

Reply:
Sounds good.  Thanks for making the changes!

Reply:
https://github.com/pytorch/pytorch/pull/13482